Contacts/app/controllers/contacts_controller.rb

```diff
@@ -0,0 +1,52 @@
+class ContactsController < ApplicationController
+    def index
+        @contacts = Contact.all
+    end
+
+    def new
+    end
+    
+    def edit
+        @contact = Contact.find_by(params[:id])
+    end
+    
+    def create
+        @contact = Contact.new
+        @contact.first_name = params[:first_name]
+        @contact.last_name = params[:last_name]
+        @contact.phone = params[:phone]
+        @contact.address = params[:address]
+        @contact.city = params[:city]
+        @contact.state = params[:state]
+        @contact.zip = params[:zip]
+        @contact.save
+        
+        redirect_to "/contacts"
+    end
+    
+    def update
+        @contact = Contact.find_by(params[:id])
+        @contact.first_name = params[:first_name]
+        @contact.last_name = params[:last_name]
+        @contact.phone = params[:phone]
+        @contact.address = params[:address]
+        @contact.city = params[:city]
+        @contact.state = params[:state]
+        @contact.zip = params[:zip]
+        @contact.save
+        
+        redirect_to "/contacts"
+    end
+
+    def search
+        @search = "%#{params[:search]}%"
+        @contacts = Contact.where("first_name LIKE ? OR last_name LIKE ?", search, search)
+    end
```

Comment:
You almost got it working! How about instead of hitting this search endpoint you hit your index action? Since you are already rendering all the contacts in your `index.html.erb`. Try moving your logic in this `search` method to your `index` then make your search form send the `search` params to the `index` method. This will hopefully get your search to work.